server/src/main/java/org/candlepin/model/AsyncJobStatus.java

```diff
@@ -152,12 +163,8 @@ public boolean isTerminal() {
     private Date endTime;
 
     @Column(name = "job_data")
-    @Type(type = "org.candlepin.hibernate.JsonSerializedDataType")
-    private Object jobData;
-
-    @Column(name = "job_result")
-    @Type(type = "org.candlepin.hibernate.JsonSerializedDataType")
-    private Object jobResult;
+    @Type(type = "org.candlepin.hibernate.DynamicJsonSerializedType")
```

Comment:
Can we use [`@Convert`](https://docs.oracle.com/javaee/7/api/javax/persistence/Convert.html) instead?

Reply:
I'm not opposed to looking into changing this over to use the JPA style of conversion, but I'm not familiar with it at all, so I can't say how that will impact things here.

Reply:
I've looked into this a bit more. The JPA AttributeConverter interface is nice and simple, but it lacks one thing: a way to make it generic. We can work around it, but it's a tad frustrating that we can't make a generic class (JsonConverter) that just accepts a class. Every different conversion needs its own unique subclass.

Reply:
I've added a commit that removes the user type in favor of the JPA converter. Using it requires more boilerplate due to the non-dynamic nature of the interface, but the amount of code in the converter is substantially reduced, and it's less janky/magical.

If we decide not to go forward with this, we can just back up the most recent commit.